### PR TITLE
I2C addon

### PR DESCRIPTION
**Pull request by WantClue**
Added I2C Diaplay functionallity it requires the pip library for Adafruit_SSD1306

config.yml does have it's own config section now for the I2C enable or disable.

used the usual I2C connections of the raspberry pi

### CODE REVIEW

File: config.yml.example [hunk 1]
```diff
@@ -50,3 +50,7 @@ influx:
   org: piaxe
   bucket: piaxe
   timezone: Europe/Berlin
+
+
+i2c:
```
Comment (shufps) [inline config.yml.example:55]: maybe should be called with a little bit less generic name like `i2c_display` :thinking:

File: piaxe/miner.py [hunk 2]
```diff
@@ -37,6 +37,7 @@
 from . import bm1366
 from . import influx
 from . import discord
+from .ssd1306 import SSD1306
```
Comment (shufps) [inline piaxe/miner.py:40]: please add a try except block or move it into the other try except block that imports raspberry specific libraries

**Comment by WantClue** [comment]
updated

File: piaxe/miner.py [hunk 3]
```diff
@@ -391,6 +393,7 @@ def __init__(self, config, address, network):
 
         self.shares = list()
         self.stats = influx.Stats()
+        self.display = SSD1306(self.stats)
```
Comment (shufps) [inline piaxe/miner.py:400]: this shouldn't be called if display is not enabled - maybe move it into the `if` that creates the thread? :thinking:

**Pull request merged**